File: polimorfismo/java/Funcionario.java
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Polimorfismo em Java
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

public abstract class Funcionario {
	private String nome;
	
	public Funcionario() {
		
	}
	
	public String getNome() {
		return nome;
	}
	
	public void setNome(String nome) {
		this.nome = nome;
	}
	
	public abstract float calcularSalarioFinal();
}

File: polimorfismo/cpp/Funcionario.cpp
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Polimorfismo em C++
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

#include "Funcionario.hpp"
#include <iostream>

using namespace std;

Funcionario::Funcionario()
{

}

string Funcionario::getNome()
{
    return nome;
}

void Funcionario::setNome(string n)
{
    nome = n;
}

float Funcionario::calcularSalarioFinal()
{

}